proto/regen/ecocredit/orderbook/v1beta1/memory.proto

```diff
@@ -19,55 +18,86 @@ message BuyOrderSellOrderMatch {
     // this index is what does the magic - it basically sorts our orders for us and more or less does the matching.
     // Basically we iterate over all bid prices within a given denom from high to low. The highest bidding orders always get filled first, and if one bid is the same price as another, the earlier bid wins - since buy order IDs are assigned incrementally we get this for free.
     // Then within a given buy order, we iterate over all the matching sell orders from the lowest to highest ask price. So the highest bids and lowest asks get filled first. If the ask price is the same, then earlier sell orders get filled first (again we get this from the sell order ID which is assigned incrementally).
-    index: { id: 1; fields: "bid_denom_id,bid_price_complement,buy_order_id,ask_price,sell_order_id";
+    index: { id: 1; fields: "market_id,bid_price_complement,buy_order_id,ask_price,sell_order_id";
       //name: "order_book_match"
     }
 
     // this index is used to delete matches once sell orders are filled
     index: { id: 2, fields: "sell_order_id" }
   };
 
-  // bid_denom_id defines the bid denom being used by the buy and sell orders. Matching always happens within a single bid denom.
-  uint32 bid_denom_id = 1;
+  // market_id defines the market within which this match exists.
+  uint64 market_id = 1;
 
   // buy_order_id is the buy order ID.
   uint64 buy_order_id = 2;
 
   // sell_order_id is the sell order ID.
   uint64 sell_order_id = 3;
 
-  // bid_price_complement is the the complement (~ operator) of the bid price encoded as a uint64 (which should have sufficient precision) - effectively ~price * 10^exponent (usually 10^6). The complement is used so that bids can be sorted high to low.
-  fixed64 bid_price_complement = 4;
+  // bid_price_complement is the the complement (^ operator) of the bid price encoded as a uint3 (which should have sufficient precision) - effectively ~price * 10^exponent (usually 10^6). The complement is used so that bids can be sorted high to low.
+  fixed32 bid_price_complement = 4;
 
-  // ask_price is the ask price encoded to a uint64. Ask prices are sorted low to high.
-  fixed64 ask_price = 5;
+  // ask_price is the ask price encoded to a uint32. Ask prices are sorted low to high.
+  fixed32 ask_price = 5;
```

Comment:
Why `fixed32` here rather than `uint32`?

Reply:
Because it's a decimal value rather than a sequence so it may use more bits than a sequence